qhub/template/stages/07-kubernetes-services/modules/kubernetes/services/argo-workflows/main.tf

```diff
@@ -0,0 +1,150 @@
+locals {
+  name = "argo-workflows"
+  argo-workflows-prefix = "argo"
+}
+
+resource "helm_release" "argo-workflows" {
+  name       = local.name
+  namespace  = var.namespace
+  repository = "https://argoproj.github.io/argo-helm"
+  chart      = "argo-workflows"
+  version    = "0.13.1"
+
+  values = concat([
+    file("${path.module}/values.yaml"),
+    # https://github.com/argoproj/argo-helm/blob/argo-workflows-0.13.1/charts/argo-workflows/values.yaml
+
+
+    jsonencode({
+      singleNamespace = true  # Restrict Argo to operate only in a single namespace (the namespace of the Helm release)
+
+      controller = {
+        metricsConfig = {
+          enabled = true  # enable prometheus
+        }
+        workflowNamespaces = [
+          "${var.namespace}"
+          ]
+        nodeSelector = {
+          "${var.node-group.key}" = var.node-group.value
+        }
+      }
+
+      server = {
+        # `sso` for OIDC/OAuth
+        extraArgs = ["--auth-mode=sso", "--insecure-skip-verify"]
+        # to enable TLS, `secure = true`
+        secure = false
+        baseHref = "/${local.argo-workflows-prefix}/"
+
+        sso = {
+          insecureSkipVerify = true
+          issuer = "https://${var.external-url}/auth/realms/${var.realm_id}"
+          clientId = {
+            name = "argo-server-sso"
+            key = "argo-oidc-client-id"
+          }
+          clientSecret = {
+            name = "argo-server-sso"
+            key = "argo-oidc-client-secret"
+          }
+          # The OIDC redirect URL. Should be in the form <argo-root-url>/oauth2/callback.
+          redirectUrl = "https://${var.external-url}/${local.argo-workflows-prefix}/oauth2/callback"
+          rbac = {
+            enabled = false
+            secretWhitelist = []
+          }
+          scopes = ["profile"]
+        }
+        nodeSelector = {
+          "${var.node-group.key}" = var.node-group.value
+        }
+      }
+
+      containerRuntimeExecutor = "emissary"
+
+    })
+  ], var.overrides)
+}
+
+resource "kubernetes_secret" "argo-oidc-secret" {
+  metadata {
+    name = "argo-server-sso"
+    namespace = var.namespace
+  }
+  data = {
+    "argo-oidc-client-id"     = module.argo-workflow-openid-client.config.client_id
+    "argo-oidc-client-secret" = module.argo-workflow-openid-client.config.client_secret
+  }
+}
+
+module "argo-workflow-openid-client" {
+  source = "../keycloak-client"
+
+  realm_id = var.realm_id
+  client_id = "argo-server-sso"
+  external-url = var.external-url
+  # role_mapping = {
```

Comment:
I made some more progress on this.  We may need to delete the insecure-skip-no-verify stuff still.  Also, in the Argo UI, we can see all the roles (jupyterhub, dask, others) instead of just argo roles.

Reply:
> We may need to delete the insecure-skip-no-verify stuff still

I think this is fine for now. Especially since we don't have a great way to set this on/off if this is a local/public deployment.

Reply:
I agree, I initially though --insecure-skip-verify had to do with authentication, but I believe it's associated with TLS instead.